### ISSUE TITLE
Can't override <Table/> "Select All" control when paginated

### ISSUE THREAD

**Issue by ChambreNoire** [ant-design/ant-design#25747]
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/antd-table-selection-ikqym)

### Steps to reproduce
I have an Ant Design table with sorting, selection and pagination. By default (if I understand it correctly), Antd Table selection works on a per-page basis. If you click the 'Select All' checkbox in the selection column header, Antd will select all the rows on the current page but if you then change page, both the rows and header will revert to a unchecked state. The selection is not lost because if you then go back to the previous page, the rows and header are all checked.

So I wanted to change this so that 'Select All' really selects all the rows of the table regardless of pagination. I found a few github issues in which recommended simply managing the selectedRowKeys array onSelect and onSelectAll functions oneself. I did this and now my selection works as expected...

However...

The "select all" checkbox in the selection column header still functions in terms of the page rows. So for example if I'm on the first page and "Select all" then deselect one of the rows, the "select all" checkbox is in the intermediate/partial state as expected. However if I then go to the next page, all rows are checked as expected and so is the "select all" checkbox in the header. For this selection to make sense functionally it would need to remain in the intermediate/partial state. It needs to reflect the selection over the entire dataset and not just the current page.

### What is expected?
To be able to override the state of the "Select all" checkbox to find a "selection over entire paginated dataset" scenario.

### What is actually happening?
Nothing. Can't find any info on how to do this!

| Environment | Info |
|---|---|
| antd | 4.4.3 |
| React | 16.11.0 |
| System | Windows 10 |
| Browser | Google Chrome 83.0.4103.116 |

---
This is a bit of a deal-breaker for us...

**Comment by cuobiezi** [comment]
Dear Team

  I guess I got this problem , I've readed the code , there are two props in `useSelection`  data & pageData, when we trigger `SELECTION_ALL` will map with data not pageData . 
  As the code design ,  map with pageData is better then map with data in `SELECTION_ALL`, Maybe this is a bug. please checked .
  If the team agree me ，please assign it to me . 

Thx & Best Regards

**Comment by 07akioni** [comment]
> Dear Team
> 
> I guess I got this problem , I've readed the code , there are two props in `useSelection` data & pageData, when we trigger `SELECTION_ALL` will map with data not pageData .
> As the code design , map with pageData is better then map with data in `SELECTION_ALL`, Maybe this is a bug. please check & confirm it.
> If the team agree me .Please assign it to me .
> 
> Thx & Best Regards

In my memory, SELECTION_ALL is trigger by a dropdown menu. Header checkbox alway reflects pageData

**Comment by cuobiezi** [comment]
> > Dear Team
> > I guess I got this problem , I've readed the code , there are two props in `useSelection` data & pageData, when we trigger `SELECTION_ALL` will map with data not pageData .
> > As the code design , map with pageData is better then map with data in `SELECTION_ALL`, Maybe this is a bug. please check & confirm it.
> > If the team agree me .Please assign it to me .
> > Thx & Best Regards
> 
> In my memory, SELECTION_ALL is trigger by a dropdown menu. Header checkbox alway reflects pageData

Yes U R right , I need to read it careful

**Comment by 07akioni** [comment]
Maybe you can add a `getHeaderCheckboxProps` for that. PR welcomed.

**Comment by JohnnyHandy** [comment]
Yes I have the same issue... as @07akioni  mentioned, a getHeaderCheckboxProps would solbe the problem I guess. Since I cant controll the check all checkbox state, I am gonna have to use another component rather than the Table one with a controlled checkbox

**Comment by JohnnyHandy** [comment]
I found this [workaround](https://stackoverflow.com/questions/61136662/how-to-change-ant-table-header-checkbox/61137799#61137799) where you can override the check all checkbox using the `columnTitle` property inside the `rowSelection` prop

**Comment by ChambreNoire** [comment]
> I found this [workaround](https://stackoverflow.com/questions/61136662/how-to-change-ant-table-header-checkbox/61137799#61137799) where you can override the check all checkbox using the `columnTitle` property inside the `rowSelection` prop

Gahh I should have thought of that! Cheers!

**Comment by afc163** [comment]
Reviewed against the latest `master` (`0b1021909f`). Table now lets consumers control the header selection checkbox.

Latest rowSelection exposes getTitleCheckboxProps and columnTitle(originalNode), covering custom checked and indeterminate state across pages.

Closing this issue accordingly. If a minimal reproduction still demonstrates different behavior on the current `master`, please open a new issue with that reproduction.
